### ISSUE TITLE
Possibility to SSR data fetched from non-page component

### ISSUE THREAD

**Issue by t-ricci-digit** [sveltejs/kit#10729]
### Describe the problem

For highly dynamic pages, when the rendered child components could change and not easily determined from the +page.svelte file, if might be useful to fetch data from child components (ie. non-page components).

Example:
   /+page.svelte <-- this component uses ChildComponent (directly or indirectly)
    |_ChildComponent.svelte <-- here I'd like to fetch data (server-side)
    
Currently, it seems not possible to await the result of the fetch in the ChildComponent (ie. will only run on the client) in SRR.

### Describe the proposed solution

It would be nice to have something like a top-level await keyword to allow the server to server side render the results of the fetch request before sending the response to the client.

```svelte
<script>const data = await fetch('...');</script>

<div>... use data ...</data>
```

### Alternatives considered

Putting all the fetches for the child components in the +page where they are used may not always be the ideal solution.
Also using the {#await} instruction in the template will only work on the client (ie. the server SSR will only show the loading text)

Next.js 13+ using AppRouter will allow ServerSide Components to use async/await to fetch data on the server (any component, not just "Pages"/routes) and render the results: https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating

### Importance

would make my life easier

### Additional Information

I would switch to Svelte + SvelteKit instead of relying on Next.js (which is way more complex and annoying sometimes, ie. use client/hooks etc.) if SvelteKit had this SSR feature to fetch on the server on non-page components too

**Comment by Conduitry** [comment]
I believe there are other issue(s) elsewhere for this, but regardless, this wouldn't work currently, for a few reasons.

Server-side rendering of a component is currently synchronous. There's an issue open in the Svelte repo about supporting top-level await (which would also require figuring out how to make SSR asynchronous), but there are a lot of things that need to be sorted out before we can make any progress on that.

Secondly, and more seriously, is the lack of asynchronous context in JS. When you run `fetch`, we need to be able to associate that back with the original incoming request that led to this outgoing request, so that we can resolve relative URLs, handle cookies, and serialize responses into the HTML so the browser doesn't need to make the same request upon hydration. JS doesn't currently provide us with the tools to determine this when there are asynchronous calls involved. This is the reason that `fetch` is an argument to the `load` functions rather than just being a global you can use anywhere.

There are some plans to, eventually, add such a feature to JS, and once that's in place and is starting to be widely supported in many environments, we will probably want to revisit the whole `load` mechanism, but it's not time to start holding your breath yet. Once this exists, there will be more of an impetus to also start thinking about asynchronous SSR.
